Editor/EditorCore/PolyShapeTool.cs

```diff
@@ -472,6 +470,16 @@ void OnOverlayGUI(UObject target, SceneView view)
             EditorGUILayout.EndVertical();
         }
 
+        internal void SetShapeHeight()
+        {
+            UndoUtility.RecordObject(polygon, "Set Height");
```

Comment:
Did some manual testing around Undo as this was touched and seems there's issues:

https://github.com/user-attachments/assets/d04c636b-8b94-403f-8ab9-aa1660070662